theseus/core/objective.py

```diff
@@ -348,15 +348,13 @@ def error(
                 for var in self.optim_vars:
                     old_data[var] = self.optim_vars[var].data
             self.update(input_data=input_data)
-        error_vector = torch.zeros(self.batch_size, self.dim()).to(
+        error_vector = torch.zeros(self.batch_size, len(self.cost_functions)).to(
```

Comment:
Does `len(self.cost_functions)` give the number (n) of cost functions? I think we want \sum_i{cost_i.dim}, where i=1,..,n.